compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs

```diff
@@ -633,6 +633,70 @@ fn coroutine_closure_to_ambiguous_coroutine<I: Interner>(
     )
 }
 
+pub(in crate::solve) fn extract_fn_def_from_const_callable<I: Interner>(
```

Comment:
i feel somewhat unhappy about the duplication with `extract_tupled_inputs_and_output_from_callable` here.

Merging these two functions seems difficult and annoying :thinking: so maybe add a comment here that this mirrors `extract_tupled_inputs_and_output_from_callable`?